android/app/src/main/java/org/chrome/device/ml/ChromeActivity.java

```diff
@@ -0,0 +1,235 @@
+/* Copyright 2020. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+package org.chrome.device.ml;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.Bundle;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import android.os.Handler;
+import android.os.IBinder;
+import android.os.Looper;
+import android.os.Message;
+import android.os.RemoteException;
+import android.util.Log;
+import android.view.View;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ScrollView;
+import android.widget.Spinner;
+import android.widget.TextView;
+import java.io.IOException;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.chrome.device.ml.service.MLService;
+import org.chrome.device.ml.service.RemoteService;
+import org.chrome.device.ml.service.RemoteServiceCallback;
+
+/** Chrome on-device ML main activity */
+public class ChromeActivity extends AppCompatActivity implements ServiceConnection {
+  private static final String TAG = "ChromeOnDeviceML";
+  private static final String [] MODELS = {"MobileBert"};
+  private static final String URL_PATH = "url_list.txt";
+  private static final int MSG_TIME_UPDATE = 1;
+
+  private Button classifyButton;
+  public TextView resultTextView;
+  private ScrollView scrollView;
+  private Spinner modelSpinner;
+  private int spinnerSelection;
+
+  // ML Service
+  private RemoteService mService;
+  private Intent mBindIntent;
+  private Handler serviceHandler;
+  private RemoteServiceCallback serviceCallback;
+
+  // Custom Tab
+  private ArrayList<String> urlList;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+    Log.v(TAG, "onCreate");
+
+    Toolbar toolbar = findViewById(R.id.toolbar);
+    setSupportActionBar(toolbar);
+
+    modelSpinner = findViewById(R.id.modelSpinner);
+    classifyButton = findViewById(R.id.button);
+    classifyButton.setOnClickListener(
+            (View v) -> {
+              buttonHandler();
```

Comment:
might be simpler to just do buttonHandler(View v) and just ignore the v so you dont have to do the lambda here

Reply:
Done